modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -145,13 +145,161 @@ class SoftMaxSubgraph : public Subgraph
+        int inpShape2 = addNodeToMatch("Shape", input);
+        int inpGather2 = addNodeToMatch("Gather", inpShape2, indices2);
+        int inpCastG2 = addNodeToMatch("Cast", inpGather2);
+        int B2 = addNodeToMatch("Constant");
+        int inpMul2 = addNodeToMatch("Mul", inpCastG2, B2);
+        int inpCastM2 = addNodeToMatch("Cast", inpMul2);
+        int inpFloor2 = addNodeToMatch("Floor", inpCastM2);
+
+        int inpUnsqueeze1 = addNodeToMatch("Unsqueeze", inpFloor1);
+        int inpUnsqueeze2 = addNodeToMatch("Unsqueeze", inpFloor2);
+        int inpConcat1 = addNodeToMatch("Concat", inpUnsqueeze1, inpUnsqueeze2);
+
+        int inpIndex = addNodeToMatch("Constant");
+
+        int inpCast3 = addNodeToMatch("Cast", inpConcat1);
+
+        int inpShape3 = addNodeToMatch("Shape", input);
+        int inpSlice = addNodeToMatch("Slice", inpShape3);
+        int inpCast4 = addNodeToMatch("Cast", inpSlice);
+
+        int inpDiv = addNodeToMatch("Div", inpCast3, inpCast4);
+        int inpConcat2 = addNodeToMatch("Concat", inpIndex, inpDiv);
+
+        addNodeToMatch("Upsample", input, inpConcat2);
+        setFusedNode("Upsample", input, indices1, B1, indices2, B2, inpIndex);
+    }
+
+    virtual bool match(const Ptr<ImportGraphWrapper>& net, int nodeId,
+                       std::vector<int>& matchedNodesIds,
+                       std::vector<int>& targetNodesIds) CV_OVERRIDE
+    {
+        if (Subgraph::match(net, nodeId, matchedNodesIds, targetNodesIds))
+            return true;
+        return false;
+    }
+
+    virtual void finalize(const Ptr<ImportGraphWrapper>&,
+                          const Ptr<ImportNodeWrapper>& fusedNode,
+                          std::vector<Ptr<ImportNodeWrapper> >& inputs) CV_OVERRIDE
+    {
+        opencv_onnx::NodeProto* node = fusedNode.dynamicCast<ONNXNodeWrapper>()->node;
+        opencv_onnx::AttributeProto* attr = node->add_attribute();
+
+        CV_Assert(node->input_size() == 6);
+
+        std::vector<opencv_onnx::NodeProto*> input_nodes(3);
+        std::vector<cv::Mat> nodes_attributes(3);
+        input_nodes[0] = inputs[5].dynamicCast<ONNXNodeWrapper>()->node; //inpIndex
+        input_nodes[1] = inputs[2].dynamicCast<ONNXNodeWrapper>()->node; //B1
+        input_nodes[2] = inputs[4].dynamicCast<ONNXNodeWrapper>()->node; //B2
+        for(int i = 0; i < 3; i++)
+        {
+            opencv_onnx::TensorProto tensor_proto = input_nodes[i]->attribute(0).t();
+            nodes_attributes[i] = getMatFromTensor(tensor_proto);
+        }
+        CV_Assert(nodes_attributes[0].total() == 2 && nodes_attributes[1].total() == 1 && nodes_attributes[2].total() == 1);
+        attr->set_name("scales");
+        attr->add_ints((int)nodes_attributes[0].at<float>(0));
+        attr->add_ints((int)nodes_attributes[0].at<float>(1));
+        attr->add_ints((int)nodes_attributes[1].at<float>(0));
+        attr->add_ints((int)nodes_attributes[2].at<float>(0));
```

Comment:
Let's match our fused layer to "Caffe2" line Upsample:

https://github.com/opencv/opencv/blob/f9442aec16484ea322b30f55479462f78f7f13df/modules/dnn/src/onnx/onnx_importer.cpp#L881-L899

So you will only need to set `height_scale` and `width_scale`.

Reply:
Done !